controllers/pagerdutyintegration/serviceorchestration_handler.go

```diff
@@ -99,50 +99,48 @@ func (r *PagerDutyIntegrationReconciler) handleServiceOrchestration(pdclient pd.
 		}
 	}
 
-	if !pdData.ServiceOrchestrationRuleApplied {
+	serviceOrchestrationConfigMap := types.NamespacedName{
+		Name:      pdi.Spec.ServiceOrchestration.RuleConfigConfigMapRef.Name,
+		Namespace: pdi.Spec.ServiceOrchestration.RuleConfigConfigMapRef.Namespace,
+	}
 
-		serviceOrchestrationConfigMap := types.NamespacedName{
-			Name:      pdi.Spec.ServiceOrchestration.RuleConfigConfigMapRef.Name,
-			Namespace: pdi.Spec.ServiceOrchestration.RuleConfigConfigMapRef.Namespace,
-		}
+	ruleConfigmap := &corev1.ConfigMap{}
 
-		ruleConfigmap := &corev1.ConfigMap{}
+	err = r.Get(context.TODO(), serviceOrchestrationConfigMap, ruleConfigmap)
+	if errors.IsNotFound(err) {
+		r.reqLogger.Info("service orchestration configmap rule not found, skip the following steps")
+		localmetrics.UpdateMetricPagerDutyServiceOrchestrationFailure(1, pdi.Name)
+		return nil
+	}
+	if err != nil {
+		return err
+	}
 
-		err = r.Get(context.TODO(), serviceOrchestrationConfigMap, ruleConfigmap)
-		if errors.IsNotFound(err) {
-			r.reqLogger.Info("service orchestration configmap rule not found, skip the following steps")
-			localmetrics.UpdateMetricPagerDutyServiceOrchestrationFailure(1, pdi.Name)
-			return nil
+	if pdiSelector.Matches(labels.Set(ruleConfigmap.GetLabels())) {
+		orchestrationConfig, err := utils.LoadConfigMapData(r.Client,
+			serviceOrchestrationConfigMap, ServiceOrchestrationDataName)
+		if err != nil {
+			return err
 		}
+
+		pdData.ServiceOrchestrationRules = orchestrationConfig
+
+		r.reqLogger.Info(fmt.Sprintf("apply the service orchestration rules from configmap: %s",
+			orchestrationConfigmapName))
+		err = pdclient.ApplyServiceOrchestrationRule(pdData)
```

Comment:
Does this mean it would happen on every reconcile for every cluster? If so, given the `ApplyServiceOrchestrationRule` function seems to itself make two calls (to get the service and to apply the orchestration), I am concerned over the potential that this might lead to potential rate limiting in our organization.

That being said, I don't have better immediate options here other than to halve the number by first retrieving the service orchestration from PD and comparing if it is different to what is defined via the configmap/pdi.

Reply:
The `ApplyServiceOrchestrationRule` will have one api call, which is HTTP PUT.

Based on my testing, it won't be triggered in the reconcile loop every time. Not sure if the `Reconcile()` return is preventing that https://github.com/openshift/pagerduty-operator/blob/master/controllers/pagerdutyintegration/pagerdutyintegration_controller.go#L236 

And also, other func in the controller is not being called everytime as well, like https://github.com/openshift/pagerduty-operator/blob/master/controllers/pagerdutyintegration/clusterdeployment_created.go#L168